### PR TITLE
Get-AzKeyVaultSecret can -AsPlainText

### LINKED ISSUE #13630: [KeyVault] Support getting plain text secret with `-AsPlainText`
## Description
Adding an optional parameter to `Get-AzKeyVaultSecret`. When specified, outputs the plain text secret.

## Design choices
- Should we return the plain text secret only, or return a `PSKeyVaultSecret` object with an extra property representing the plain text secret?
    - Per below sggestions, we'll just return the secret text.
- Any other cmdlets also need `-AsPlainText`?
    - Not this time. 

## Expectation

It should work in
- [x] PowerShell 7.0, 7.1
- [x] PowerShell core
- [x] Windows PowerShell 5.1
- [x] [ConstrainedLanguage mode](https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_language_modes?view=powershell-7.1#constrained-language-constrained-language)
- [x] Linux
- [x] MacOS

## Cost
s

### PR DESCRIPTION
**Pull request by isra-fel**
## Description

Fix #13630

If you Get-AzKeyVaultSecret -AsPlainText , it will just return a plain text secret string, instead of a secret object.

## Checklist

- [x] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [x] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
        - {Please put the link here}
    - [x] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

File: src/KeyVault/KeyVault.Test/PesterTests/Secret.Tests.ps1 [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+. ../Scripts/Common.ps1
+$secretName = Get-SecretName
+$secretText = 'dummy text'
+$secretTextV2 = 'dummy text 2'
+Set-AzKeyVaultSecret -VaultName $vaultName -Name $secretName -SecretValue (ConvertTo-SecureString $secretText -AsPlainText -Force)
+Set-AzKeyVaultSecret -VaultName $vaultName -Name $secretName -SecretValue (ConvertTo-SecureString $secretTextV2 -AsPlainText -Force)
+
+Describe "Get secret" {
+    It "should write secrets in plain text if -AsPlainText" {
```
Comment (isra-fel) [inline src/KeyVault/KeyVault.Test/PesterTests/Secret.Tests.ps1:12]: It's a data-plane API, so I wrote a liveonly test here

File: src/KeyVault/KeyVault/Commands/RemoveAzureKeyVaultSecret.cs [hunk 2]
```diff
@@ -22,9 +22,6 @@
 
 namespace Microsoft.Azure.Commands.KeyVault
 {
-    [GenericBreakingChange("If you have soft-delete protection enabled on this key vault, this secret will be moved to the soft deleted state. " +
```
Comment (isra-fel) [inline src/KeyVault/KeyVault/Commands/RemoveAzureKeyVaultSecret.cs:25]: This should have been removed in last major release

File: src/KeyVault/KeyVault/Commands/GetAzureKeyVaultSecret.cs [hunk 3]
```diff
@@ -209,7 +218,7 @@ public override void ExecuteCmdlet()
             if (!string.IsNullOrEmpty(Version))
             {
                 secret = DataServiceClient.GetSecret(VaultName, Name, Version);
-                WriteObject(secret);
+                WriteSecret(secret);
```
Comment (wyunchi-ms) [inline src/KeyVault/KeyVault/Commands/GetAzureKeyVaultSecret.cs:221]: Should we keep the same behaiour for the `deletedSecret` in line 241? I think it's reasonable.

Comment (isra-fel) [reply src/KeyVault/KeyVault/Commands/GetAzureKeyVaultSecret.cs:221]: `deletedSecret` does not contain secret value, it's just metadata, so it is fine.

**Review by wyunchi-ms — APPROVED** [review]

**Pull request merged**